lib: date_time: Automatically setting time to Zephyr and modem (#6125)
nrfconnect/sdk-nrf: doc/nrf/libraries/others/date_time.rst

The :c:func:`date_time_set` function can be used to obtain the current date-time information from external sources independent of the internal date-time update routine.
Time from GPS can be such an external source.

> The option :kconfig:`CONFIG_DATE_TIME_AUTO_UPDATE` determines whether the obtained date-time information is triggered automatically when the LTE network becomes available.

Author: Please check.

The :c:func:`date_time_set` function can be used to obtain the current date-time information from external sources independent of the internal date-time update routine.
Time from GPS can be such an external source.

The option :kconfig:`CONFIG_DATE_TIME_AUTO_UPDATE` determines whether date-time update is triggered automatically when the LTE network becomes available.
Libraries that require date-time information can just enable this library to get updated time information.

> Applications don't need to do anything to trigger time update when application starts because this library handles it automatically.

Reviewer: ```suggestion
Applications do not need to do anything to trigger time update when they start because this library handles it automatically.
```

> Current date-time information is stored as Zephyr time when it has been retrieved and hence applications can also get the time using POSIX function `clock_gettime`.
> Current date-time information is also stored as modem time if modem doesn't have valid time.

Reviewer: ```suggestion
Current date-time information is stored as Zephyr time when it has been retrieved and hence applications can also get the time using the POSIX function ``clock_gettime``.
It is also stored as modem time if the modem does not have valid time.
```

To get date-time information from the library, either call the :c:func:`date_time_uptime_to_unix_time_ms` function or the :c:func:`date_time_now` function.
See the API documentation for more information on these functions.

.. note::

>    It is recommended to set :kconfig:`CONFIG_DATE_TIME_AUTO_UPDATE` option to trigger a time update when device has connected to LTE.
>    If application has time dependent operations immediately after connecting to LTE network, it should wait for information that time has been updated.
>    If :kconfig:`CONFIG_DATE_TIME_AUTO_UPDATE` is not set, the first date-time update cycle (after boot) does not occur until the time set by the :kconfig:`CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS` has elapsed.

Reviewer: ```suggestion
   It is recommended to set the :kconfig:`CONFIG_DATE_TIME_AUTO_UPDATE` option to trigger a time update when the device has connected to LTE.
   If an application has time-dependent operations immediately after connecting to LTE network, it should wait for a confirmation telling that time has been updated.
   If the :kconfig:`CONFIG_DATE_TIME_AUTO_UPDATE` option is not set, the first date-time update cycle (after boot) does not occur until the time set by the :kconfig:`CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS` option has elapsed.
```

   Configure this option to control the frequency with which the library fetches the time information.

:kconfig:`CONFIG_DATE_TIME_AUTO_UPDATE`

>    Configure this option to enable updating of the system and modem time based on obtained time information.

Author: Basically copy-pasted from previous comment

   Configure this option to control the frequency with which the library fetches the time information.

> :kconfig:`CONFIG_DATE_TIME_AUTO_UPDATE`
> 
>    Configure this option to determine whether date-time update is triggered automatically when device has connected to LTE.

Reviewer: ```suggestion
:kconfig:`CONFIG_DATE_TIME_AUTO_UPDATE`

CONFIG_DATE_TIME_AUTO_UPDATE - Trigger date-time update when LTE is connected
   Configure this option to determine whether date-time update is triggered automatically when the device has connected to LTE.
```

> 

Reviewer: Maybe an entry to the changelog is needed.

Author: Done
